src/parser/meta.rs

```diff
@@ -47,17 +32,27 @@ pub fn parse_meta_get_status(buf: &[u8]) -> IResult<&[u8], Response> {
     ))(buf)
 }
 
-// TODO: refactor this into individual parse_meta_*_response methods
+pub fn parse_meta_get_response(buf: &[u8]) -> Result<Option<(usize, Response)>, ErrorKind> {
+    let bufn = buf.len();
+    let result = parse_meta_get_data_value(buf);
+
+    match result {
+        Ok((left, response)) => {
+            let n = bufn - left.len();
```

Comment:
Can we make these variables more explicit? It's hard to tell what is going on here.

Reply:
Changed some names a little bit which hopefully makes it a bit more clear, but the code path is unfortunately a bit obscure by nature.